spec/ordinary-and-exotic-object-behaviours.html

```diff
@@ -0,0 +1,316 @@
+<!DOCTYPE html>
+<meta charset="utf8" />
+
+<emu-clause id="sec-ordinary-and-exotic-objects-behaviours">
+  <h1>Ordinary and Exotic Objects Behaviours</h1>
+
+  <emu-clause id="sec-built-in-exotic-object-internal-methods-and-slots">
+    <h1>Built-in Exotic Object Internal Methods and Slots</h1>
+    <emu-clause id="sec-record-exotic-objects">
+      <h1>Record Exotic Objects</h1>
+      <p>A Record object is an exotic object that encapsuates a Record value and exposes virtual string-indexed data properties corresponding to the individual keys set on the underlying Record value. All keys properties are non-writable and non-configurable.</p>
+
+      <p>An object is a <dfn id="record-exotic-object">Record exotic object</dfn> (or simply, a Record object) if its following internal methods use the following implementations. These methods are installed in RecordCreate.</p>
+
+      <p>Record exotic objects have the same internal slots as ordinary objects. They also have a [[RecordData]] internal slot.</p>
+
+      <emu-clause id="sec-record-exotic-objects-setprototypeof-v">
+        <h1>[[SetPrototypeOf]] ( _V_ )</h1>
+        <p>When the [[SetPrototypeOf]] internal method of a Record exotic object _R_ is called with argument _V_, the following steps are taken:</p>
+        <emu-alg>
+          1. Return ! SetImmutablePrototype(_R_, _V_).
+        </emu-alg>
+      </emu-clause>
+      
+      <emu-clause id="sec-record-exotic-objects-isextensible">
+        <h1>[[IsExtensible]] ()</h1>
+        <p>When the [[IsExtensible]] internal method of a Record exotic object _R_ is called, the following steps are taken:</p>
+        <emu-alg>
+          1. Return *false*.
+        </emu-alg>
+      </emu-clause>
+
+      <emu-clause id="sec-record-exotic-objects-getownproperty-p">
+        <h1>[[GetOwnProperty]] ( _P_ )</h1>
+        <p>When the [[GetOwnProperty]] internal method of a Record exotic object _R_ is called with property key _P_, the following steps are taken:</p>
+        <emu-alg>
+          1. Assert: IsPropertyKey(_P_) is *true*.
+          1. If Type(_P_) is a Symbol, return *undefined*.
+          1. Let _getResult_ be ! RecordGet(_R_, _P_).
+          1. If _getResult_ is *undefined*, return *undefined*.
```

Comment:
It's important to distinguish undefined from missing here, as the value in the record may actually be undefined. You can use `~empty~` as a distinguished return value of RecordGet for this case, for example.